File: src/broadcast_switch.py
import subprocess
import sys

def get_crontab():
    try:
        result = subprocess.run(['crontab', '-l'], capture_output=True, text=True, check=True)
        return result.stdout.splitlines()
    except subprocess.CalledProcessError as e:
        print(f"Cannot read crontab: {e.stderr.strip()}")
        sys.exit(1)

def update_crontab(cron_jobs):
    new_cron = "\n".join(cron_jobs)
    if new_cron and not new_cron.endswith('\n'):
        new_cron += '\n'
    process = subprocess.Popen(['crontab'], stdin=subprocess.PIPE, text=True)
    process.communicate(input=new_cron)
    process.wait()

def main():
    cron_jobs = get_crontab()
    if sys.argv[1] == '--start':
        cron_jobs = [task.lstrip('#') for task in cron_jobs]
    elif sys.argv[1] == '--stop':
        cron_jobs = [f'#{task}' if not task.startswith('#') else task for task in cron_jobs]
    update_crontab(cron_jobs)
    print("Crontab changed.")

if __name__ == "__main__":
    main()

File: src/broadcast_switch.php
<?php

function getCrontab() {
    exec('crontab -l 2>&1', $output, $return_var);
    if ($return_var !== 0) {
        echo "Cannot read crontab: " . implode("\n", $output);
        exit(1);
    }
    return $output;
}

function updateCrontab($cronJobs) {
    $newCron = implode("\n", $cronJobs);
    if (!empty($newCron) && substr($newCron, -1) !== "\n") {
        $newCron .= "\n";
    }
    $process = proc_open('crontab', [
        ['pipe', 'r'],
        ['pipe', 'w'],
        ['pipe', 'w'],
    ], $pipes);

    if (is_resource($process)) {
        fwrite($pipes[0], $newCron);
        fclose($pipes[0]);
        stream_get_contents($pipes[1]);
        fclose($pipes[1]);
        stream_get_contents($pipes[2]);
        fclose($pipes[2]);
        proc_close($process);
    }
    echo "Crontab changed.\n";
}

function startBC() {
    $cronJobs = getCrontab();
    $newJobs = array_map(function($job) {
        return ltrim($job, '#');
    }, $cronJobs);
    updateCrontab($newJobs);
}

function stopBC() {
    $cronJobs = getCrontab();
    $newJobs = array_map(function($job) {
        return (strpos($job, '#') === false) ? '#' . $job : $job;
    }, $cronJobs);
    updateCrontab($newJobs);
}

?>
